finagle-core/src/main/scala/com/twitter/finagle/tracing/Trace.scala

```diff
@@ -48,19 +49,20 @@ object Trace {
     "com.twitter.finagle.tracing.TraceContext"
   ) {
     private val local = new ThreadLocal[Array[Byte]] {
-      override def initialValue(): Array[Byte] = new Array[Byte](32)
+      override def initialValue(): Array[Byte] = new Array[Byte](40)
     }
 
     def marshal(id: TraceId): Buf =
       Buf.ByteArray.Owned(TraceId.serialize(id))
 
     /**
      * The wire format is (big-endian):
-     *     ''spanId:8 parentId:8 traceId:8 flags:8''
+     *     ''spanId:8 parentId:8 traceId:8 flags:8 (traceIdHigh:8)''
      */
     def tryUnmarshal(body: Buf): Try[TraceId] = {
```

Comment:
Not part of your change, but it's really disappointing that this is essentially a duplicate of `TraceId.deserialize` other than the bytes representation the function receives. Other than we cache a 40 byte thread-local Array[Byte] in this method, we could trivially forward it to the `TraceId.deserialize` function. 😞 This should be fixed later.